[bug]: asterisk.c fails to compile when HAVE_LIBEDIT_IS_UNICODE isn't defined

### Severity

Major

### Versions

all

### Components/Modules

asterisk.c

### Operating Environment

CentOS 7 or any older distro where libedit isn't compiled for unicode.

### Frequency of Occurrence

Constant

### Issue Description

asterisk.c:2728
```c
	if (!ast_opt_exec && fds[1].revents) {
#if HAVE_LIBEDIT_IS_UNICODE
		num_read = editline_read_char(editline, cp);
		if (num_read < 1) {
			break;
		}
#else
		char c = '\0';

		num_read = read(STDIN_FILENO, &c, 1);
		if (num_read < 1) {
			break;
		}

		*cp = CHAR_TO_LIBEDIT(c);

#endif
		return num_read;
	}
```

The `#if` on line 2729 should be an `#ifdef`.

### Relevant log output

```shell
asterisk.c: In function 'ast_el_read_char':
asterisk.c:2729:5: error: "HAVE_LIBEDIT_IS_UNICODE" is not defined [-Werror=undef]
 #if HAVE_LIBEDIT_IS_UNICODE
     ^
cc1: all warnings being treated as errors
```

### Asterisk Issue Guidelines

- [x] Yes, I have read the Asterisk Issue Guidelines

--- a/main/asterisk.c
+++ b/main/asterisk.c
@@ -2726,7 +2726,7 @@ static int ast_el_read_char(EditLine *editline, CHAR_T_LIBEDIT *cp)
 		}
 
 		if (!ast_opt_exec && fds[1].revents) {
-#if HAVE_LIBEDIT_IS_UNICODE
+#ifdef HAVE_LIBEDIT_IS_UNICODE
 			num_read = editline_read_char(editline, cp);
 			if (num_read < 1) {
 				break;

